Description: Use system brotli on Unix non-portable builds
 PR merged to RC2, but not yet present on RC1. This fixes the nativeaot-sb autopkgtest.
Author: Andy Gocke <andy@commentout.net>
Origin: upstream, https://github.com/dotnet/runtime/pull/107225
Forwarded: not-needed
Last-Update: 2024-09-26
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: dotnet9-9.0.100-9.0.0~rc1/src/runtime/eng/DotNetBuild.props
===================================================================
--- dotnet9-9.0.100-9.0.0~rc1.orig/src/runtime/eng/DotNetBuild.props	2024-09-06 09:52:40.000000000 -0300
+++ dotnet9-9.0.100-9.0.0~rc1/src/runtime/eng/DotNetBuild.props	2024-09-26 13:09:43.533554045 -0300
@@ -88,7 +88,7 @@
 
       <!-- Handle system libraries -->
       <UseSystemLibs Condition="'$(UseSystemLibs)' != ''">+$(UseSystemLibs)+</UseSystemLibs>
-      <InnerBuildArgs Condition="$(UseSystemLibs.Contains('+brotli+'))">$(InnerBuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_BROTLI=true</InnerBuildArgs>
+      <InnerBuildArgs Condition="'$(PortableBuild)' != 'true'">$(InnerBuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_BROTLI=true</InnerBuildArgs>
       <InnerBuildArgs Condition="$(UseSystemLibs.Contains('+libunwind+'))">$(InnerBuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_LIBUNWIND=true</InnerBuildArgs>
       <!-- TODO: llvm-libunwind -->
       <!-- TODO: LinuxTracepoints -->
Index: dotnet9-9.0.100-9.0.0~rc1/src/runtime/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
===================================================================
--- dotnet9-9.0.100-9.0.0~rc1.orig/src/runtime/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets	2024-09-26 13:09:00.083321058 -0300
+++ dotnet9-9.0.100-9.0.0~rc1/src/runtime/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets	2024-09-26 13:09:12.169818772 -0300
@@ -168,6 +168,11 @@
       <NativeSystemLibrary Include="crypto" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(UseSystemBrotli)' != 'false' and Exists('$(IlcSdkPath)nonportable.txt')">
+      <NativeSystemLibrary Include="brotlienc" />
+      <NativeSystemLibrary Include="brotlidec" />
+    </ItemGroup>
+
     <ItemGroup Condition="'$(StaticOpenSslLinking)' == 'true' and '$(NativeLib)' != 'Static'">
       <NativeLibrary Include="$(IntermediateOutputPath)libs/System.Security.Cryptography.Native/build/libSystem.Security.Cryptography.Native.OpenSsl.a" />
       <DirectPInvoke Include="libSystem.Security.Cryptography.Native.OpenSsl" />
